File: ch4/jskim/money.java
class Dollar
{
    private int amount;
    
    Dollar(int amount)
    {
         this.amount = amount;
    }

    void times(int multiplier)
    {
        return new Dollar(amount * multiplier);
    }

    public boolean equals(Object object)
    {
        Dollar dollar = (Dollar) object;
        
        return amount == dollar.amount;
    }
}

public void testEquality()
{
    assertTrue(new Dollar(5).equals(new Dollar(5)));

    assertFalse(new Dollar(5).equals(new Dollar(6)));
}

public void testMultiplication()
{
    Dollar five = new Dollar(5);
    
    assertEquals(new Dollar(10), five.times(2));

    assertEquals(new Dollar(15), five.times(3));
}

File: ch4/jskim/money.py
import unittest

class Dollar:
    
    __amount:int

    def __init__(self, amount:int):
        self.__amount = amount

    def times(self, multiplier:int):
        return Dollar(self.__amount * multiplier)
    
    def equals(self, dollar:object):
        return self.__amount == dollar.__amount

class testEquality(unittest.TestCase):
    
    def testEquality(self):
        self.assertTrue(Dollar(5).equals(Dollar(5)))

        self.assertFalse(Dollar(5).equals(Dollar(6)))

class testMultiplication(unittest.TestCase):

    def testMultiplication(self):
        five = Dollar(5)

        self.assertTrue(Dollar(10).equals(five.times(2)))

        self.assertTrue(Dollar(15).equals(five.times(3)))
    
if __name__ == "__main__":
    unittest.main()